examples/with-route-as-modal/pages/post/[postId].js

```diff
@@ -1,11 +1,36 @@
 import { useRouter } from 'next/router'
+import Modal from 'react-modal'
 import Post from '../../components/Post'
+import PostCardGrid, { posts } from '../../components/PostCardGrid'
 
-const PostPage = () => {
+Modal.setAppElement('#__next')
+
+const PostPage = ({ postId }) => {
   const router = useRouter()
-  const { postId } = router.query
 
-  return <Post id={postId} />
+  return (
+    <>
+      <Modal
+        isOpen={true} // The modal should always be shown on page load, it is the 'page'
+        onRequestClose={() => router.push('/')}
+        contentLabel="Post modal"
+      >
+        <Post id={postId} pathname={router.pathname} />
+      </Modal>
+      <PostCardGrid />
+    </>
+  )
 }
 
 export default PostPage
+
+export function getStaticProps({ params: { postId } }) {
+  return { props: { postId: postId } }
+}
+
+export function getStaticPaths() {
+  return {
+    paths: posts.map((postId) => ({ params: { postId: postId.toString() } })),
+    fallback: false,
+  }
+}
```

Comment:
This changes should go to a new page, otherwise the current use case of the example will be replaced, right?

Reply:
Not really, but I see your point -- I can add a different page like `/article/[articleId]` for the dynamic routing use case if you prefer

The current use case is rendering `index.js` and opening a modal when the query matches, not actually navigating to `/post/[postId]` (the queryString routing approach), hence the `pathname` is `/` 

I would also call out that in the current [code the "gallery" or grid view is missing from that "page"](https://github.com/zeit/next.js/blob/canary/examples/with-route-as-modal/pages/post/%5BpostId%5D.js) if it were to be shared as a link.  This addresses that UX, but to your point, it might be a confusing mixture of two approaches

Reply:
Went ahead and made these changes 👆